web/packages/api/src/forKusama.ts

```diff
@@ -0,0 +1,800 @@
+    if (!destParachain) {
+        throw Error(`Could not find ${destParaId} in the asset registry.`)
+    }
+
+    const sourceAssetMetadata = registry.parachains[sourceParaId].assets[tokenAddress.toLowerCase()]
+    if (!sourceAssetMetadata) {
+        throw Error(`Token ${tokenAddress} not registered on source asset hub.`)
+    }
+    const destAssetMetadata =
+        registry.kusama?.parachains[destParaId].assets[tokenAddress.toLowerCase()]
+    if (!destAssetMetadata) {
+        throw Error(`Token ${tokenAddress} not registered on destination asset hub.`)
+    }
+
+    return { sourceAssetMetadata, destAssetMetadata, sourceParachain }
+}
+
+export async function getDeliveryFee(
+    sourceAssetHub: ApiPromise,
+    direction: Direction,
+    registry: AssetRegistry,
+    defaultFee?: bigint
+): Promise<DeliveryFee> {
+    const feeStorageKey = xxhashAsHex(":XcmBridgeHubRouterBaseFee:", 128, true)
+    const feeStorageItem = await sourceAssetHub.rpc.state.getStorage(feeStorageKey)
+    let leFee = new BN((feeStorageItem as Codec).toHex().replace("0x", ""), "hex", "le")
+
+    let defaultFeeConfiguredInRuntime: bigint
+    if (direction == Direction.ToPolkadot) {
+        defaultFeeConfiguredInRuntime = 10_602_492_378n // .0106KSM
+    } else {
+        defaultFeeConfiguredInRuntime = 333_794_429n // 0.033 DOT
+    }
+    let xcmBridgeFee: bigint
+    if (leFee.eqn(0)) {
+        console.warn("Asset Hub onchain XcmBridgeHubRouterBaseFee not set. Using default fee.")
+        xcmBridgeFee = defaultFee ?? defaultFeeConfiguredInRuntime
+    } else {
+        xcmBridgeFee = BigInt(leFee.toString())
+    }
+
+    let forwardedXcm = buildTransferToKusamaExportXCM(
+        sourceAssetHub.registry,
+        DOT_LOCATION,
+        dotLocationOnKusamaAssetHubLocation(),
+        erc20Location(registry.ethChainId, "0x0000000000000000000000000000000000000000"), // actual token location doesn't matter here, just weighing the message
+        xcmBridgeFee,
+        xcmBridgeFee,
+        registry.assetHubParaId,
+        340282366920938463463374607431768211455n,
+        "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "0x0000000000000000000000000000000000000000000000000000000000000000"
+    )
+
+    let bridgeHubDeliveryFee = await calculateDeliveryFee(
+        sourceAssetHub,
+        registry.bridgeHubParaId,
+        forwardedXcm
+    )
+
+    // In either DOT or KSM
+    let totalFee = xcmBridgeFee + bridgeHubDeliveryFee
+
+    let totalFeeInDot = 0n
+    // Convert KSM to DOT
+    if (direction == Direction.ToPolkadot) {
+        console.info("Converting KSM to DOT")
+        let amount = xcmBridgeFee + bridgeHubDeliveryFee
+        totalFeeInDot = await quoteFeeSwap(
+            sourceAssetHub,
+            { parents: 1, interior: "Here" },
+            { parents: 2, interior: { x1: [{ GlobalConsensus: { Polkadot: null } }] } },
+            amount
+        )
```

Comment:
Well, for the Kusama->Polkadot direction, as discussed earliar, the fee asset is KSM - a runtime parameter declared [here](https://github.com/polkadot-fellows/runtimes/blob/9859c5f95c9031526c4dbe728e3f95b262327d63/system-parachains/asset-hubs/asset-hub-kusama/src/xcm_config.rs#L485), I'm not sure swapping it to DOT here make sense.

Reply:
I think we need to decide what the fee asset for Kusama to Polkadot should be. I kept it DOT because it is more consistent with the rest of the bridge, but I don't have strong feelings about this. WDYT @alistair-singh @vgeddes?

Reply:
I'd assume the bridge fee in `T::FeeAsset` [here](https://github.com/paritytech/polkadot-sdk/blob/ff5c67c8a3d60ccb3f0af114a07db8f4bec485a0/bridges/modules/xcm-bridge-hub-router/src/lib.rs#L330) is always the native token of the source chain.  It’s taken from the end user and sent to the Treasury on the source chain, meaning the bridge transfer can’t proceed without it.